mcmetadata/content.py

```diff
@@ -215,6 +225,40 @@ def extract(self, url: str, html_text: str, include_metadata: bool = False):
         }
 
 
+class LxmlExtractor(AbstractExtractor):
+    """
+    An inefficient fallback extractor that should work in all contexts
+    """
+    
+    def extract(self, url: str, html_text: str, include_metadata: bool = False):
+        cleaner = Cleaner(scripts=True, javascript=True, 
+                comments=True, style=True, links=True, meta=True,
+                add_nofollow=False, page_structure=True, 
+                processing_instructions=True, embedded=True,
+                frames=True, forms=True, annoying_tags=True, 
+                allow_tags=[None],safe_attrs_only=False)
+
+
+        parsed = lxml.etree.HTML(html_text)
+        #This ensures that the etree is cast correctly-
+        #It seems like pages with multiple head nodes sometimes confuse
+        #the parser without this step
+        circular = fromstring(tostring(parsed))
+        content_string = tostring(cleaner.clean_html(circular))
+        
+        self.content = {
+            "url": url,
+            'text': content_string,
+            'title': None,
+            'potential_publish_date': None,
+            'top_image_url': None,
+            'authors': None,
+            'extraction_method': METHOD_LXML,
+        }
+        print(self.content)
```

Comment:
@pgulley you forgot a `print` statement. Please use `logger.debug` if you want to have logging.

Reply:
(✖╭╮✖)
should be patched now!